docs/tasksDir/pod-security-with-kubewarden.md

```diff
@@ -0,0 +1,356 @@
+    app: nginx
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: template-nginx
+  template:
+    metadata:
+      labels:
+        app: template-nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:1.14.2
+        securityContext:
+          runAsNonRoot: true
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+              - "ALL"
+          seccompProfile:
+            type: "RuntimeDefault"
+
+        ports:
+        - containerPort: 80
+EOF
+```
+
+This time the Pod Security Admission will allow the Pod to be created, but the Pod creation will still fail.
+
+```console
+kubectl get pods -n my-namespace
+NAME                                READY   STATUS                       RESTARTS   AGE
+nginx-deployment-57d8568bbb-h4bx7   0/1     CreateContainerConfigError   0          47s
+
+```
+
+That's happening because the container definition didn't specify which user should
+be used when starting a program inside of the container. That results with the
+the root user (`0`) being used, which is not allowed by the `runAsNonRoot` directive:
+
+```console
+kubectl get pods -n my-namespace nginx-deployment-57d8568bbb-h4bx7 -o json | jq ".status.containerStatuses"
+[
+  {
+    "image": "nginx:1.14.2",
+    "imageID": "",
+    "lastState": {},
+    "name": "nginx",
+    "ready": false,
+    "restartCount": 0,
+    "started": false,
+    "state": {
+      "waiting": {
+        "message": "container has runAsNonRoot and image will run as root (pod: \"nginx-deployment-57d8568bbb-8mvkc_my-namespace(add7bcc5-3d23-43d0-94e9-6e78f887a53f)\", container: nginx)",
+        "reason": "CreateContainerConfigError"
+      }
+    }
+  }
+]
+```
+
+This is where Kubewarden can help. In this case the 
+[`user-group-policy`]
+(https://artifacthub.io/packages/kubewarden/user-group-psp/user-group-psp)
+policy can be used to mutate the Deployment definition, configuring a default
+user to be used to all the containers that omit this information.
+Or even enforcing a user ID range. Let's apply the policy:
+The policy even allows to set a user ID range:
+
+:::note
+This example assumes the Kubewarden stack is already installed inside of your
+Kubernetes cluster.
+See the [QuickStart](../quick-start.md) article
+for more details.
+:::
+
+```console
+kubectl apply -f - <<EOF
+apiVersion: policies.kubewarden.io/v1
+kind: ClusterAdmissionPolicy
+metadata:
+  name: user-group-psp
+spec:
+  policyServer: default
+  module: registry://ghcr.io/kubewarden/policies/user-group-psp:latest
+  rules:
+  - apiGroups: [""]
+    apiVersions: ["v1"]
+    resources: ["pods"]
+    operations:
+    - CREATE
+    - UPDATE
+  mutating: true
+  settings:
+    run_as_user:
+      rule: "MustRunAs"
+      overwrite: false
+      ranges:
+        - min: 1000
+          max: 2000
+        - min: 4000
+          max: 5000
+    run_as_group:
+      rule: "RunAsAny"
+    supplemental_groups:
+      rule: "RunAsAny"
+EOF
+```
+
```

Comment:
maybe we should let the user know he has to wait for the policy to be active before moving to the next step

Reply:
done with d54a12124be4c50167b1ba742a663484b7123267